EAR implementation

9sneha-n opened the pull request:
### :pushpin: References

-   **Issue:** Closes https://app.clickup.com/t/865crk2a4

### :memo: Implementation



### :art: Screenshots
![image](https://github.com/EyeSeeTea/glass-dev/assets/83749675/901ca6b6-5df5-4961-80ed-34ab22e21061)
![image](https://github.com/EyeSeeTea/glass-dev/assets/83749675/e82ca29b-26e2-458a-9589-24427a653dba)
![image](https://github.com/EyeSeeTea/glass-dev/assets/83749675/be136cc8-65a0-4bab-bb76-ba8affa8a25d)
### :fire: Testing
For EAR testing :
EAR_NonConfidential:EAR_NonConfidential@123 - Should be able to see only non-confidential or approved signals
EAR_Confidential_Global:EAR_Confidential_Global@123 - Should be able to see all signals of all countries
EAR_Confidential_KOR:EAR_Confidential_KOR@123 - Should be able to see all signals of Korea + all non-confidential signals

anagperal reviewed (approved):
thanks @9sneha-n you've done a lot of great work 👏 I have a few comments on the code, from my point of view:

- thanks a lot for the comments along the code explaining some parts step by step, not only in this PR but in general.  They have helped me to follow and understand it better.
- in `useSignals.ts`  in the useEffect maybe the ifs can be encapsulated in separate functions whose name is descriptive of what it does because I found it a bit harder to follow
- maybe this is something you had already planned to do in the future, but in `NewSignalForm.tsx` maybe a hook could be created

ifoche reviewed (changes requested):
Oh, @deeonwuli and btw. After merging with development I never get the list of signals loaded. Could you also have a look pulling the latest changes? Thank you!

9sneha-n commented:
> thanks @9sneha-n you've done a lot of great work 👏 I have a few comments on the code, from my point of view:
> 
> * thanks a lot for the comments along the code explaining some parts step by step, not only in this PR but in general.  They have helped me to follow and understand it better.
> * in `useSignals.ts`  in the useEffect maybe the ifs can be encapsulated in separate functions whose name is descriptive of what it does because I found it a bit harder to follow
> * maybe this is something you had already planned to do in the future, but in `NewSignalForm.tsx` maybe a hook could be created

@anagperal  Thanks for the detailed review of this super long PR and for the great feedback. I have implemented your review comments.

9sneha-n commented:
> Functional testing:
> 
> * EAR shouldn't be producing any card in the homepage requesting to submit yearly data as it's based on signals and not on data submissions
>   ![screenshots_1691496857_1414_08082023_1530x542](https://user-images.githubusercontent.com/10931645/259096431-8b98e951-8b34-4fa4-afde-9dc9ff48e06d.png)
> * dev.user in EAR admin user group but not in confidential group tries to save a draft and gets this:
>   ![screenshots_1691497279_1421_08082023_1248x852](https://user-images.githubusercontent.com/10931645/259096404-98a4a473-f859-48e3-a376-42b88f4bfc34.png)
>   The problem is that EAR program is not assigned to the country. We could show a meaningful message like "Country not assigned to EAR program, please contact your administrator" or something like that
> * Creating a non-confidential draft after assigning the program to the country says success but I cannot see the signal in the signals list. However I can see 2 calls to signals where the signal does appear, even if I access with a user that is assigned to the country. If we add the confidential group then drafts appear. But we probably would need to allow users with capture access from the country to see their own signals in draft. Could we eliminate the restriction to be in the confidential group for those cases?
> * Edition of drafts seems not to be implemented yet so you cannot complete a draft after created it.

Hi @ifoche , thank you for the review. I have implemented all the feedback

- No EAR card on  homepage
<img width="1728" alt="Screenshot 2023-09-06 at 9 20 34 PM" src="https://github.com/EyeSeeTea/glass-dev/assets/83749675/72a429ca-2797-4ab2-a544-04b6187ef002">

- Updated Error message
<img width="1728" alt="Screenshot 2023-09-06 at 9 19 47 PM" src="https://github.com/EyeSeeTea/glass-dev/assets/83749675/fcba8227-bb52-4fee-8cf2-43d47aad3173">

- User with non confidential user group will see draft signals for all OU it has access to.
<img width="1728" alt="Screenshot 2023-09-06 at 9 23 47 PM" src="https://github.com/EyeSeeTea/glass-dev/assets/83749675/5a62db44-c647-42b1-be8f-3644f9954a11">

-  Implemented Edit and publish of draft signals
<img width="1728" alt="Screenshot 2023-09-06 at 9 26 47 PM" src="https://github.com/EyeSeeTea/glass-dev/assets/83749675/e3db7756-e291-45b4-8985-01b090348a82">

ifoche reviewed (approved):
thanks @9sneha-n
